dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -315,8 +310,14 @@ Aggregator::Aggregator(
                 "Aggregation does not support spill because aggregator hash table does not support two level");
         }
     }
-    if (register_operator_spill_context != nullptr && agg_spill_context->isSpillEnabled())
+    if (register_operator_spill_context != nullptr)
         register_operator_spill_context(agg_spill_context);
+    if (agg_spill_context->isSpillEnabled())
+    {
+        /// for aggregation, the input block is sorted by bucket number
+        /// so it can work with MergingAggregatedMemoryEfficientBlockInputStream
+        agg_spill_context->buildSpiller(header);
```

Comment:
```suggestion
        agg_spill_context->buildSpiller(getHeader(false));
```

Reply:
Why need to change this?

Reply:
The code can be simplified, no need to initialize `Block header` in L302

Reply:
Yes, you are right, updated the code.